src/relay/pass/mac_count.cc

```diff
@@ -0,0 +1,138 @@
+   * \param call_node The CONV 2D call node.
+   * \return The number of MACs.
+   */
+  int64_t ComputeConv2DMacs(const CallNode* call_node) {
+    CHECK(IsConv2DNode(call_node))
+        << "The input call node must be a CONV 2D node.";
+    if (!call_node->checked_type_.defined()) {
+      LOG(WARNING) << "The infer type pass should be called before the mac count pass";
+      return 0;
+    }
+    Array<Expr> args = call_node->args;
+    CHECK(args.size() == 2)
+        << "The number of input arguments of a CONV 2D node should be 2.";
+    const auto* expr0 = args[0]->checked_type().as<TensorTypeNode>();
+    Array<IndexExpr> shape0 = expr0->shape;
+    CHECK(shape0.size() == 4)
+        << "The dimension of an input tensor to Dense node should be 4.";
+    int64_t input_channel = static_cast<int64_t>(shape0[1].as<IntImm>()->value);
+    const auto* conv_2d_attr = call_node->attrs.as<Conv2DAttrs>();
+    Array<IndexExpr> kernel_size = conv_2d_attr->kernel_size;
+    CHECK(kernel_size.size() == 2)
+        << "The dimension of the kernel size in Conv 2D should be 2.";
+    const auto* expr = call_node->checked_type().as<TensorTypeNode>();
+    Array<IndexExpr> output_tensor = expr->shape;
+    CHECK(output_tensor.size() == 4)
+        << "The dimension of the output tensor in Conv 2D should be 4.";
+    int64_t count = input_channel * GetCartesianProd(output_tensor) * GetCartesianProd(kernel_size);
+    return count;
+  }
+
+  /*
+   * \brief Get the number of MACs of a Dense node.
+   * \param call_node The Dense call node.
+   * \return The number of MACs.
+   */
+  int64_t ComputeDenseMacs(const CallNode* call_node) {
+    CHECK(IsDenseNode(call_node))
+        << "The input call node must be a Dense node.";
+    if (!call_node->checked_type_.defined()) {
+      LOG(WARNING) << "The infer type pass should be called before the mac count pass";
+      return 0;
+    }
+    Array<Expr> args = call_node->args;
+    CHECK(args.size() == 2)
+        << "The number of input arguments of a Dense node should be 2.";
+    const auto* expr0 = args[0]->checked_type().as<TensorTypeNode>();
+    const auto* expr1 = args[1]->checked_type().as<TensorTypeNode>();
+    Array<IndexExpr> shape0 = expr0->shape;
+    Array<IndexExpr> shape1 = expr1->shape;
+    CHECK(shape0.size() == 2 && shape1.size() == 2)
+        << "The dimension of an input tensor to Dense node should be 2.";
+    int64_t d1 = static_cast<int64_t>(shape0[0].as<IntImm>()->value);
+    int64_t d2 = static_cast<int64_t>(shape0[1].as<IntImm>()->value);
+    int64_t d3 = static_cast<int64_t>(shape1[0].as<IntImm>()->value);
+    int64_t d4 = static_cast<int64_t>(shape1[1].as<IntImm>()->value);
+    CHECK(d2 == d4)
+        << "The dimensions of input arguments do not match.";
+    int64_t count = d1 * d2 * d3;
+    return count;
+  }
+
+  int64_t GetCartesianProd(Array<IndexExpr> arr) {
+    int64_t ret = 1;
+    for (int i = 0; i < arr.size(); i++) {
```

Comment:
`int -> size_t` since `arr.size()` is `size_t` or probably just `for (const auto& it : arr)`